Lesson27/Merve/update.js

```diff
@@ -0,0 +1,51 @@
+const urlParams = new URLSearchParams(window.location.search);
+const postId = urlParams.get('postId');
+
+// 1. Fetch post data for the post with given postId
+function getPostById(id) {
+  return fetch(`${URL}${id}`)
+    .then(response => response.json())
+    .then(data => data)
+    .catch(error => console.error('Error fetching post:', error));
+}
+
+
+
+// 2. Prefill form with post title and post body
+getPostById(postId).then(post => {
+  document.getElementById('postId').value = post.id;
+  document.getElementById('postContent').value = post.content;
+});
+
+
+
+// 3. Select form element and attach 'submit' event listener to it, send new post information with UPDATE request
+document.getElementById('update-post-form').addEventListener('submit', function (event) {
+  event.preventDefault();
+
+  const postId = document.getElementById('postId').value;
+  const postContent = document.getElementById('postContent').value;
+
+  updatePost(postId, postContent);
```

Comment:
Pass data to the `updatePost` function in the object format so you have both title and body.
```suggestion
  updatePost(postId, {title: postTitle, body: postContent});
```